tests/suite/foundations/calc.typ

```diff
@@ -259,3 +259,30 @@
 --- calc-max-uncomparable ---
 // Error: 16-19 cannot compare 1pt with 1em
 #calc.max(1em, 1pt)
+
+--- calc-norm ---
+#test(calc.norm2(1, 2, -3, 0.5), calc.sqrt(14.25))
+#test(calc.norm2(3cm, 4cm), 5cm)
+#test(calc.norm2(3em, 4em), 5em)
+#test(calc.norm2(), 0)
+#test(calc.pnorm(3, 1, 2), calc.pow(9, 1/3))
+
+--- calc-norm-expect-number ---
+// Error: 16-19 expected a number
+#calc.norm2(1, 1in)
+
+--- calc-norm-expect-abs ---
+// Error: 18-21 expected an absolute length
+#calc.norm2(1in, 2em)
+
+--- calc-norm-expect-em ---
+// Error: 18-21 expected an em
+#calc.norm2(1em, 2in)
+
+--- calc-norm-expect-abs-or-em ---
+// Error: 13-20 expected an absolute length or em
+#calc.norm2(1cm+1em)
+
+--- calc-norm-expect-number-or-length ---
+// Error: 13-16 expected a number or length
+#calc.norm2(10%)
```

Comment:
Most files end with a new line (as configured in `.editorconfig`), so it would be better to add a new empty line below this line.